### DISCUSSION TITLE
Limits or quotas

### DISCUSSION THREAD

**Discussion by sjpritchard** [speechly/speechly#312] [Q&A] [answered]
Are there any quotas/limits on the maximum number of intents, entities and utterances allowed in an app or project? I couldn't find anything in the documentation.

**Comment by bigdatabaracus** [comment, marked as answer]
@sjpritchard no maximum limitations on number of intents or entities. I think there is a maximum size for the configuration. @teelisyys what was the maximum size of the configuration, again?

**Reply by teelisyys** [reply]
The CSV imports have a maximum size of 60000 lines. The overall configuration has size limit based on its complexity, as even short configuration can be expanded to a complex set of rules. Unfortunately this means that those limits cannot be clearly formulated.
